### ISSUE TITLE
[ECS] ASG Capacity Provider Does Not Scale with Blue/Green Deploys

### ISSUE THREAD

**Issue by e-moshaya** [aws/containers-roadmap#675]
I have an ECS Cluster with ASG Capacity Provider as well as blue/green deploys enabled. However, when I initiate a blue/green deploy via codedeploy and there is not enough capacity left on the EC2 instance, the codedeploy deployment fails due to insufficient capacity on the EC2 instance. Moreover, the capacity provider does not do its job and scale out the ASG desired capacity to cater for codedeploy jobs.

Event message for the ECS service states ```service configuration was unable to place a task because no container instance met all of its requirements. The closest matching container-instance 33398145d2f442dbb20b57c964a03e6c has insufficient memory available. For more information, see the Troubleshooting section.```

**Comment by kkopachev** [comment]
I hit the same issue with `rolling update` deployment services as well. Pre-existing services with service autoscaling in place does not affect capacity provider metric at all. I figured service has to be recreated and then it automatically gets `Capacity provider strategy` field (and loses EC2/Fargate radios). 
That might be good info to add somewhere in documentation that existing services has to be recreated.

However, it is impossible to create `Blue/Green` deployment service in cluster with enabled capacity provider. I was not able to find that in any docs either.

**Comment by coultn** [comment]
@kkopachev You can update a service using the rolling update controller to use capacity providers.  However you are correct that blue/green is not supported at this time.

**Comment by regevbr** [comment]
@kkopachev I have recreated the service and the field appeared, but still I get the same error and not auto scaling of ec2 instances...

See my detailed explanation here - https://github.com/aws/containers-roadmap/issues/76#issuecomment-578439351

**Comment by davi7461** [comment]
any updates on when can expect this to be solved?

**Comment by tovbinm** [comment]
This is so badly needed!

**Comment by slwinsett** [comment]
Any update? This is affecting us as well.

**Comment by alexandergunnarson** [comment]
From https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-type-bluegreen.html:
> Cluster capacity providers are not supported when using the blue/green deployment type.

Given how much of a game changer ECS cluster autoscaling and blue/green deploys are individually, it would be amazing to be able to use them together. Does the team have an estimate as to time of completion? Thank you! 🙏

**Comment by jg-edt** [comment]
This appears to be supported now according to the link that @alexandergunnarson shared above:

> Amazon ECS service auto scaling is supported when using the blue/green deployment type. When a new deployment is started, if there are scaling activities occurring then CodeDeploy will wait 5 minutes before starting the deployment. If the scaling activities complete within 5 minutes, the deployment will continue. If the scaling activities do not complete within 5 minutes, the deployment will fail.

https://docs.aws.amazon.com/AmazonECS/latest/userguide/deployment-type-bluegreen.html

**Comment by MartinLoeper** [comment]
Please implement this as we are apparently unable to use Fargate Spot capacity otherwise (which would save us $$$).

**Comment by srrengar** [comment]
Closing this in favor of #713

**Comment by ADNANFAKHAR4504** [comment]
please correct me if I'm wrong 
I think In ecs blue/green deployment, we can't use capacity providers (fargate spot)
